src/renderer/captureTool/js/capture.js

```diff
@@ -0,0 +1,1014 @@
+                value: gpii.captureTool.determineDefaultPrefsSet(arg.preferences)});
+            transaction.commit();
+        });
+
+        // Get initial keyin state and solutions metadata
+        ipcRenderer.send("getInstalledSolutions", "Please please!");
+        ipcRenderer.send("modelUpdate");
+    };
+
+    gpii.captureTool.loadTemplate = function (templateName) {
+        var resolvedPath = require("path").join(__dirname, "html", templateName + ".handlebars");
+        var finalTemplate = require("electron").remote.require("fs").readFileSync(resolvedPath) + "";    //require(resolvedPath);
+        return finalTemplate;
+    };
+
+    gpii.captureTool.startCapture = function (that) {
+        that.applier.change("currentPage", "3_capturing_settings");
+        that.render("3_capturing_settings");
+
+        var captureAnimationLoop = function () {
+            var curCount = that.model.captureAnimationLoopTimer + 1;
+            fluid.each(that.dom.locate("captureAnimationLoopTimer"), function (el) {
+                el.innerText = curCount;
+            });
+            fluid.each(that.dom.locate("captureAnimationLoopEllipsis"), function (el) {
+                if (curCount % 3 === 0) {
+                    el.innerText = ". . .";
+                }
+                else if ((curCount + 1) % 3 === 0) {
+                    el.innerText = ". .";
+                }
+                else {
+                    el.innerText = ".";
+                }
+            });
+            that.applier.change("captureAnimationLoopTimer", curCount);
+
+            if (that.model.captureDone && curCount > 5) {
+                that.selectAllSettingsToKeepButton();
+                that.render("3_what_to_keep");
+                // This needs to happen again, because the page hasn't rendered yet,
+                // and therefore the html element doens't exist yet.
+                that.updateNumSettingsSelected();
+                that.updateSolutionSettingsTree();
+                that.applier.change("captureDone", false);
+                that.applier.change("captureAnimationLoopTimer", 1);
+            }
+            else {
+                setTimeout(captureAnimationLoop, 1000);
+            }
+
+
+        };
+        setTimeout(captureAnimationLoop, 1000);
+
+        var options = {};
+        if (that.model.whatToCapture === "running") {
+            options.solutionsList = Object.keys(that.model.runningSolutions);
+        }
+        else if (that.model.whatToCapture === "chooseapps") {
+            options.solutionsList = that.model.solutionsToCapture;
+        }
+
+        // Clear any previous capture
+        that.applier.change("capturedSettings", {}, "DELETE");
+
+        ipcRenderer.send("getAllSolutionsCapture", options);
+    };
+
+    gpii.captureTool.nextButton = function (that, currentPage) {
+        if (currentPage === "1_sign_in") { // Currently not used with UX redesign
```

Comment:
Pretty regrettable hardcoded logic, but I don't see a great reason to improve it. We're not about to implement...... Spring Web Flow

Reply:
I don't know... I bet it would be much easier these days to implement spring web flow with our sequencing stuff :p